web/src/message_list.js

```diff
@@ -112,6 +113,7 @@ export class MessageList {
             render_info = this.append_to_view(bottom_messages, opts);
         }
 
+        narrow.handle_empty_narrow_message_home();
```

Comment:
Calling the function here instead of removing `this.narrowed` to avoid issues such as the narrow banner not appearing when unmuting a topic within a stream.